crates/iota-data-ingestion-core/src/tests.rs

```diff
@@ -90,6 +90,24 @@ impl Worker for TestWorker {
     }
 }
 
+/// This worker implementation always returns an error when processing a
+/// checkpoint.
+///
+/// Useful for testing graceful shutdown logic and behaviours
+#[derive(Clone)]
+struct TestWorkerFaulty;
```

Comment:
Maybe we can just rename this to `FaultyWorker`. It's already under the `test` scope so that would be fine.

Reply:
@samuel-rufi [fixed](https://github.com/iotaledger/iota/pull/5496/commits/43731e15728a3f79b505e823ec06f4c19af75194)